azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/implementation/AppServiceBaseImpl.java

```diff
@@ -0,0 +1,263 @@
+    }
+
+    @Override
+    public void verifyDomainOwnership(String certificateOrderName, String domainVerificationToken) {
+        verifyDomainOwnershipAsync(certificateOrderName, domainVerificationToken).toBlocking().subscribe();
+    }
+
+    @Override
+    public Observable<Void> verifyDomainOwnershipAsync(String certificateOrderName, String domainVerificationToken) {
+        IdentifierInner identifierInner = new IdentifierInner().withIdentifierId(domainVerificationToken);
+        identifierInner.withLocation("global");
+        return this.manager().inner().webApps().createOrUpdateDomainOwnershipIdentifierAsync(resourceGroupName(), name(), certificateOrderName, identifierInner)
+                .map(new Func1<IdentifierInner, Void>() {
+                    @Override
+                    public Void call(IdentifierInner identifierInner) {
+                        return null;
+                    }
+                });
+    }
+
+    @Override
+    public void start() {
+        this.manager().inner().webApps().start(resourceGroupName(), name());
+        refresh();
+    }
+
+    @Override
+    public void stop() {
+        this.manager().inner().webApps().stop(resourceGroupName(), name());
+        refresh();
+    }
+
+    @Override
+    public void restart() {
+        this.manager().inner().webApps().restart(resourceGroupName(), name());
+        refresh();
+    }
+
+    @Override
+    public void swap(String slotName) {
+        this.manager().inner().webApps().swapSlotWithProduction(resourceGroupName(), name(), new CsmSlotEntityInner().withTargetSlot(slotName));
+        refresh();
+    }
+
+    @Override
+    public void applySlotConfigurations(String slotName) {
+        this.manager().inner().webApps().applySlotConfigToProduction(resourceGroupName(), name(), new CsmSlotEntityInner().withTargetSlot(slotName));
+        refresh();
+    }
+
+    @Override
+    public void resetSlotConfigurations() {
+        this.manager().inner().webApps().resetProductionSlotConfig(resourceGroupName(), name());
+    }
+
+    @SuppressWarnings("unchecked")
+    public FluentImplT withNewAppServicePlan(String name) {
+        appServicePlan = (AppServicePlanImpl) this.manager().appServicePlans().define(name).withRegion(regionName());
+        String id = ResourceUtils.constructResourceId(this.manager().subscriptionId(),
+                resourceGroupName(), "Microsoft.Web", "serverFarms", name, "");
+        inner().withServerFarmId(id);
+        return (FluentImplT) this;
+    }
+
+    public FluentImplT withFreePricingTier() {
+        return withPricingTier(AppServicePricingTier.FREE_F1);
+    }
+
+    @SuppressWarnings("unchecked")
+    public FluentImplT withPricingTier(AppServicePricingTier pricingTier) {
+        appServicePlan = appServicePlan
+                .withPricingTier(pricingTier);
+        if (super.creatableGroup != null && isInCreateMode()) {
+            appServicePlan = appServicePlan.withNewResourceGroup(resourceGroupName());
```

Comment:
Should be `appServicePlan = appServicePlan.withNewResourceGroup(super.creatableGroup);` right?